### DISCUSSION TITLE
Bot not working through hosting

### DISCUSSION THREAD

**Discussion by Happyethan8** [eritislami/evobot#893] [Q&A] [answered]
Getting a long mess of errors after "node index.js" The bot will send me an affirmative msg in console it is turning on and act normally in Discord up until it's time for music to actually play. I'm hosting through DigitalOcean and using Bitvise.
![2021-09-16_15-22-30](https://user-images.githubusercontent.com/24516275/133687637-734092a1-99c0-4284-8b65-f5ca4c81cd61.png)
.

**Comment by Happyethan8** [comment]
It was working fine while I was running it locally on VScode. I think it is having troubles downloading all the dependencies. This is the first major error msgs it throws while "npm install". Not sure how to edit my OP or comment without marking it as Solved. But it is for sure not as of now.
![2021-09-16_15-30-44](https://user-images.githubusercontent.com/24516275/133688449-1d7801a1-c832-4afd-964a-8e6a1a0fca31.png)

**Comment by eritislami** [comment, marked as answer]
I also have my bot on digitalocean and it's working fine, obviously the last error you posted is not an issue from the bot because its a 404 error from github, I'm not sure what node.js version you are using but I recommend 14.15.5 for best compatibility. 
Also [PM2](https://pm2.keymetrics.io/) is great for handling node processes.

**Reply by eritislami** [reply]
Here's what it looks like when i do `npm ci` which removes node_modules and installs fresh packages
https://i.imgur.com/NSMOQnu.mp4

**Reply by Happyethan8** [reply]
I was on 14.15.5 I ended up manually putting it to the latest 16v and added it to pm2. I'm assuming the later part is what fixed it. Thank you. I guess node index.js was only keeping the bot working while the terminal was open and as soon as I closed the terminal I think it was stopping. PM2 must be fixing that some how. Thanks again for the speedy response I wish I would of noticed it earlier lol

**Comment by Happyethan8** [comment]
So I manually updated node.js to 16 and added opus manually. I also noticed that the bot would only play music while I had the terminal opening so I connected it with PM2 Monit and now it's all working as intended I believe.
